Book/php7/extensions_design/hooks.rst

```diff
@@ -350,4 +350,73 @@ overwrite of ``zend_interrupt_function``.
 Replacing Opcode Handlers
 *************************
 
-TODO
+It is possible to override the Zend engine's individual handlers for opcodes. This might be useful
+to ignore the ``@`` operator or count how often each opcode gets executed. The API in the engine
+is only geared up for having *one* extension defined handler per opcode, which makes it important
+that you as extension author, take care of handlers already set by other extensions.
+
+The basic APIs in the engine are::
+
+    void zend_set_user_opcode_handler(int opcode, user_opcode_handler_t handler);
+    user_opcode_handler_t zend_get_user_opcode_handler(int opcode);
+    
+The `user_opcode_handler_t` is a function pointer, and each handler has the following signature[1]_::
+
+    int my_handler(zend_execute_data *execute_data);
+
+The handler's return value is important, and a few constants are defined that have a meaning:
+
+ZEND_USER_OPCODE_CONTINUE
+    Execute next opcode
+ZEND_USER_OPCODE_RETURN
+    Exit from executor (return from function)
+ZEND_USER_OPCODE_DISPATCH
+    Call original opcode handler
+ZEND_USER_OPCODE_ENTER
+    Enter into new op_array without recursion
+ZEND_USER_OPCODE_LEAVE
+    Return to calling op_array within the same executor
```

Comment:
I don't think this is going to format properly.

Reply:
It's called a definition list, and it formats just fine → https://github.com/derickr/PHP-Internals-Book/blob/patch-1/Book/php7/extensions_design/hooks.rst#replacing-opcode-handlers

Reply:
TIL. As you can see, I don't use rst much...

Reply:
I do → https://github.com/derickr/derickrethans-articles :-)